src/rest.cpp

```diff
@@ -67,6 +67,23 @@ static bool RESTERR(HTTPRequest* req, enum HTTPStatusCode status, std::string me
     return false;
 }
 
+/**
+ * Get the node context.
+ *
+ * @param[in]  req  the HTTP request, whose status code will be set to NOT_FOUND
+ *                  if node context is not found
+ * return pointer to the node context or nullptr if not found.
+ */
+static NodeContext* GetNodeContext(const util::Ref& context, HTTPRequest* req)
+{
+    NodeContext* node = context.Has<NodeContext>() ? &context.Get<NodeContext>() : nullptr;
+    if (!node) {
+        RESTERR(req, HTTP_NOT_FOUND, "node context not found");
```

Comment:
I think this should be 500 or 422? I don't agree with "node context not found" as this is not what the client is requesting.

Reply:
This code path can never be executed, unless there is a programming error. I am happy to change the error string if there is a better suggestion, but I don't think it matters much, because it is dead code.

Reply:
Right, meaning its 500 internal server error :trollface:

Reply:
@promag makes a convincing argument that this should be a 500 internal server error :)

Reply:
thx, replaced with "500"